src/drf_pydantic/parse.py

```diff
@@ -94,8 +94,13 @@ def _convert_field(field: pydantic.fields.ModelField) -> serializers.Field:
         extra_kwargs["required"] = True
     else:
         extra_kwargs["required"] = field.required
+    if field.field_info.extra.get("allow_blank") is not None:
+        extra_kwargs["allow_blank"] = field.field_info.extra.get("allow_blank")
     if field.default is not None:
         extra_kwargs["default"] = field.default
+        # If the default value is blank then we must set allow_blank to True
+        if field.default == "":
+            extra_kwargs["allow_blank"] = True
```

Comment:
Let's move the new logic to the code block where we deal with strings, because this is a feature exclusive to `CharField` and we can have default for bools as well, which don't have this option in DRF.

https://github.com/georgebv/drf-pydantic/blob/80a8394cbbb1c978270a93a2a62bf6694f925c2a/src/drf_pydantic/parse.py#L122-L127

Reply:
I understand that bool will never be equal to `""` (unlike in some - looking at `js` - languages) but it would make more sense to check for this in the context of adding string-related options.

Reply:
I meant that you compared arbitrary type (all we knew was that `field.default is not None`) to an empty string. In JavaScript things like `"" == []` return true, but luckily not in Python (which is strongly typed). Nevertheless, we still should do that comparison only when we constraint types (which you now do).